api/s2n.h

```diff
@@ -687,6 +687,266 @@ extern int s2n_cert_get_utf8_string_from_extension_data_length(const uint8_t *ex
+
+/**
+ * Gets the negotiated PSK identity length from the s2n connection object. The negotiated PSK 
+ * refers to the chosen PSK by the server to be used for the connection. 
+ * 
+ * This API can be used to determine if the negotiated PSK exists. If negotiated PSK exists a 
+ * call to this API returns a value greater than zero. If the negotiated PSK does not exist, the 
+ * value `0` is returned.
+ * 
+ * @param conn A pointer to the s2n_connection object that successfully negotiated a PSK connection.
+ * @param identity_length The length of the negotiated PSK identity. 
+ */
+S2N_API 
+int s2n_connection_get_negotiated_psk_identity_length(struct s2n_connection *conn, uint16_t *identity_length);
+
+/**
+ * Gets the negotiated PSK identity from the s2n connection object. 
+ * If the negotiated PSK does not exist, the PSK identity will not be obtained and no error will be returned. 
+ * Prior to this API call, use `s2n_connection_get_negotiated_psk_identity_length` to determine if a 
+ * negotiated PSK exists or not. 
+ *
+ * # Safety
+ *
+ * The negotiated PSK identity will be copied into the identity buffer on success.
+ * Therefore, the identity buffer must have enough memory to fit the identity length.
+ * 
+ * @param conn A pointer to the s2n_connection object.
+ * @param identity The negotiated PSK identity obtained from the s2n_connection object. 
+ * @param max_identity_length The maximum length for the PSK identity. If the negotiated psk_identity length is 
+ * greater than this `max_identity_length` value an error will be returned.
+ */
+S2N_API 
+int s2n_connection_get_negotiated_psk_identity(struct s2n_connection *conn, uint8_t *identity, uint16_t max_identity_length);
+
+struct s2n_offered_psk;
+
+/**
+ * Creates a new s2n offered PSK object. 
+ * An offered PSK object represents a single PSK sent by the client.
+ * 
+ * # Safety
+ * 
+ * Use `s2n_offered_psk_free` to free the memory allocated to the s2n offered PSK object created by this API. 
+ *
+ * @return struct s2n_offered_psk* Returns a pointer to the newly created offered PSK object.
+ */
+S2N_API 
+struct s2n_offered_psk* s2n_offered_psk_new();
+
+/**
+ * Frees the memory associated with the `s2n_offered_psk` object.
+ *
+ * @param psk A pointer to the `s2n_offered_psk` object to be freed.
+ */
+S2N_API 
+int s2n_offered_psk_free(struct s2n_offered_psk **psk);
+
+/**
+ * Gets the PSK identity and PSK identity length for a given offered PSK object. 
+ * 
+ * @param psk A pointer to the offered PSK object being read.
+ * @param identity The PSK identity being obtained.
+ * @param size The length of the PSK identity being obtained.
+ */
+S2N_API 
+int s2n_offered_psk_get_identity(struct s2n_offered_psk *psk, uint8_t** identity, uint16_t *size);
+
+struct s2n_offered_psk_list;
+
+/**
+ * Checks whether the offered PSK list has an offered psk object next in line in the list.
+ * An offered PSK list contains all the PSKs offered by the client for the server to select.
```

Comment:
We should explicitly say that if this returns false, we are at the end of the list.

Reply:
Your previous comment said it was repetitive. I just got rid of `If an offered PSK exists next in line, this API returns True, otherwise False.` :/